lib/internal/streams/pipeline.js

```diff
@@ -147,6 +153,63 @@ async function pump(iterable, writable, finish, { end }) {
   }
 }
 
+async function pumpToWeb(readable, writable, finish, { end }) {
+  let error;
+  let onresolve = null;
+
+  if (isTransformStream(writable)) {
+    writable = writable.writable;
+  }
+
+  const resume = (err) => {
+    if (err) {
+      error = err;
+    }
+
+    if (onresolve) {
+      const callback = onresolve;
+      onresolve = null;
+      callback();
+    }
+  };
+
+  const wait = () => new Promise((resolve, reject) => {
+    if (error) {
+      reject(error);
+    } else {
+      onresolve = () => {
+        if (error) {
+          reject(error);
+        } else {
+          resolve();
+        }
+      };
+    }
+  });
+
+  eos(writable, resume);
+
+  try {
+    const writer = writable.getWriter();
+    for await (const chunk of readable) {
+      const written = writer.write(chunk);
+      if (!written) {
+        await wait();
```

Comment:
How and when does this wait resume?

Reply:
hmm yes this part is wrong writer.write will be promise 
mistake 😅😅